packages/core/src/processor_doc.ts

```diff
@@ -0,0 +1,33 @@
+//
+// Copyright © 2021 Anticrm Platform Contributors.
+//
+// Licensed under the Eclipse Public License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License. You may
+// obtain a copy of the License at https://www.eclipse.org/legal/epl-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+import type { Doc, Emb } from './classes'
+import core from './component'
+import type { Tx, TxAddCollection, TxCreateDoc } from './tx'
+
+export class TxProcessor {
+  async tx (tx: Tx): Promise<void> {
+    switch (tx._class) {
+      case core.class.TxCreateDoc:
+        return await this.txCreateDoc(tx as TxCreateDoc<Doc>)
+      case core.class.TxAddCollection:
+        return await this.txAddCollection(tx as TxAddCollection<Emb>)
+    }
+    return await Promise.resolve()
+  }
+
+  protected async txCreateDoc (tx: TxCreateDoc<Doc>): Promise<void> {}
+  protected async txAddCollection (tx: TxAddCollection<Emb>): Promise<void> {}
+}
```

Comment:
OK, why this is moved out of `storage`? Why new file, there is already good place for this such as `storage` or `tx`

Reply:
Moved to tx.